### PR TITLE
libguestfs: 1.40.2 -> 1.44.1

### PR DESCRIPTION
**Pull request by erdnaxe**
###### Motivation for this change

Current libguestfs is incompatible with QEMU 6.1: https://github.com/NixOS/nixpkgs/pull/135877#issuecomment-907109266

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libguestfs/default.nix [hunk 1]
```diff
@@ -12,19 +12,19 @@ assert javaSupport -> jdk != null;
...
 
   src = fetchurl {
-    url = "https://libguestfs.org/download/1.40-stable/${pname}-${version}.tar.gz";
-    sha256 = "ad6562c48c38e922a314cb45a90996843d81045595c4917f66b02a6c2dfe8058";
+    url = "https://libguestfs.org/download/1.44-stable/${pname}-${version}.tar.gz";
+    sha256 = "09dhmlbfdwirlmkasa28x69vqs5xndq0lnng6b4if76s6bfxrdvj";
   };
 
-  nativeBuildInputs = [ autoreconfHook makeWrapper pkg-config ];
+  nativeBuildInputs = [ autoreconfHook bison flex getopt makeWrapper pkg-config ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libguestfs/default.nix:22]: Can we set strictDeps?

File: pkgs/development/libraries/libguestfs/default.nix [hunk 2]
```diff
@@ -12,19 +12,19 @@ assert javaSupport -> jdk != null;
 
 stdenv.mkDerivation rec {
   pname = "libguestfs";
-  version = "1.40.2";
+  version = "1.44.1";
 
   src = fetchurl {
-    url = "https://libguestfs.org/download/1.40-stable/${pname}-${version}.tar.gz";
-    sha256 = "ad6562c48c38e922a314cb45a90996843d81045595c4917f66b02a6c2dfe8058";
+    url = "https://libguestfs.org/download/1.44-stable/${pname}-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libguestfs/default.nix:18]:
```suggestion
    url = "https://libguestfs.org/download/${lib.versions.majorMinor version}-stable/${pname}-${version}.tar.gz";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Review by r-burns — COMMENTED** [review]
To fix libguestfs-with-appliance, add `"--with-guestfs-path=${placeholder "out"}/lib/guestfs"` to configureFlags.

File: pkgs/development/libraries/libguestfs/default.nix (hunk 1 above)
Comment (erdnaxe) [reply pkgs/development/libraries/libguestfs/default.nix]: I just enabled strictDeps and moved cdrkit, cpio, gperf, OCaml, Perl to nativeBuildInputs to make it build. I fear some other dependencies in `buildInputs` should go in `nativeBuildInputs`.

**Comment by alyssais** [comment]
Can we use fetchpatch now that your patch has been applied upstream?

**Comment by erdnaxe** [comment]
Result of `nixpkgs-review pr 135900` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>diffoscope</li>
    <li>libguestfs</li>
    <li>libguestfs-with-appliance</li>
    <li>python38Packages.guestfs</li>
    <li>python39Packages.guestfs</li>
    <li>vagrant</li>
  </ul>
</details>

**Review by r-burns — APPROVED** [review]
Big kudos for upstreaming the patch :)

Result of `nixpkgs-review pr 135900` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>diffoscope</li>
    <li>libguestfs</li>
    <li>libguestfs-with-appliance</li>
    <li>python38Packages.guestfs</li>
    <li>python39Packages.guestfs</li>
    <li>vagrant</li>
  </ul>
</details>

**Comment by bjornfor** [comment]
FYI, this upgrade effectively removed `virt-v2v, virt-p2v and the boot analysis tools`, because upstream factored those out to separate repos since libguestfs-1.42.1, ref https://www.libguestfs.org/guestfs-release-notes-1.42.1.html.

**Comment by bjornfor** [comment]
> FYI, this upgrade effectively removed `virt-v2v, virt-p2v and the boot analysis tools`, because upstream factored those out to separate repos since libguestfs-1.42.1, ref https://www.libguestfs.org/guestfs-release-notes-1.42.1.html.

WIP PR to bring back `virt-v2v`: https://github.com/NixOS/nixpkgs/pull/235432

**Pull request merged**